src/server/AppManager.ts

```diff
@@ -758,15 +758,15 @@ export class AppManager {
         }
 
         if (enable) {
-            const externalComponent = this.externalComponentManager.getExternalComponent(app.getID());
+            const externalComponents = this.externalComponentManager.getExternalComponents(app.getID());
 
             this.commandManager.registerCommands(app.getID());
-            this.externalComponentManager.registerComponent(app.getID(), externalComponent, true);
+            this.externalComponentManager.registerExternalComponents(app.getID(), externalComponents);
```

Comment:
Why are you passing in the external components to register when the `externalComponentManager` already has those stored in its instance?

Reply:
no @graywolf336 . before this executed, the `externalComponentManager` does not have any external components in `providedExternalComponents`. The `addExternalComponent` method only add the external component to `apTouchedExternalComponents`, and `registerExternalComponent` will actually add them to `providedExternalComponents` during an app enabling period.

Reply:
![image](https://user-images.githubusercontent.com/32427260/66145122-bb851c80-e63c-11e9-90a6-3bfd28ba406f.png)

Reply:
Right, I understand that. But I think we should encapsulate the logic for the external components inside of the class `AppExternalComponentManager`. So, instead of that class having the method with the signature of `registerExternalComponents(appId: string, externalComponents: Map<string, IExternalComponent>): void` why not have it be just `registerExternalComponents(appId: string): void` and then pull from the variable `appTouchedExternalComponents` in that class?

Reply:
The better practice 👍